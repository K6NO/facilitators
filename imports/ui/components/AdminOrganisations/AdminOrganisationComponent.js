/* eslint-disable max-len, no-return-assign */

import React from 'react';
import PropTypes from 'prop-types';
import { ButtonGroup, Button, Row, Col } from 'reactstrap';
import { Meteor } from 'meteor/meteor';
import { Bert } from 'meteor/themeteorchef:bert';
import Loading from '../../components/Loading/Loading';
import DeleteOrganisationButton from './DeleteOrganisationButton';

import './AdminOrganisationComponent.scss';


class AdminOrganisationComponent extends React.Component {
  constructor(props){
    super(props);
    this.state = {
        organisation : {
          name: this.props.organisation && this.props.organisation.name || '',
          country: this.props.organisation && this.props.organisation.country || '',
          active: this.props.organisation && this.props.organisation.active || false,
        },
        orgActiveState: this.props.organisation.active || false,
        editing: false,
      };
  }

  switchEditing = () => {
    this.setState({
        ...this.state,
        editing: !this.state.editing,
    });
  }

  switchOrganisationState(value) {
      this.setState({
          ...this.state,
          organisation : {
              ...this.state.organisation,
              active: value,
          },
          orgActiveState : value,
      }, () => {
        this.saveOrganisation('active', this.state.orgActiveState);
      }); 
  }

  updateOrganisationState = (field, event) => {
    this.setState({
      organisation : {
        ...this.state.organisation,
        [field]: event.target.value,
      }
    });
  }

  saveOrganisation = (stateKey, newValue) => {
    const { organisation } = this.props;
    if(organisation[stateKey] !== newValue) {
        organisation[stateKey] = newValue;

        ['createdAt', 'updatedAt', 'owner'].forEach(e => delete organisation[e]);
  
        // save chapter
        Meteor.call('organisations.update', organisation, (error) => {
            if (error) {
            Bert.alert(error.reason, 'danger');
            console.error(error);
            } else { 
            Bert.alert(organisation.name + ' saved', 'success');
            }
        });
    }
  }

  render() {
    const { loading } = this.props;
        return ( !loading ? (
        <tr>
            <td>
                {this.state.editing ? (
                    <input name="name" type="text" 
                    onBlur={(e) => this.saveOrganisation('name', e.target.value)}
                    onChange={(e) => this.updateOrganisationState('name', e)} 
                    value={this.state.organisation.name} />
                ) : (
                    <span>{this.state.organisation.name}</span> 
                )}
            </td>
            <td>
                {this.state.editing ? (
                <input name="country" type="text" 
                onBlur={(e) => this.saveOrganisation('billingCountry', e.target.value)}
                onChange={(e) => this.updateOrganisationState('billingCountry', e)} 
                value={this.state.organisation.billingCountry} />
                ) : (
                    <span>{this.state.organisation.billingCountry}</span> 
                )}
            </td>
            <td>
                {this.state.editing ? (
                    <input name="city" type="text" 
                    onBlur={(e) => this.saveOrganisation('billingCity', e.target.value)}
                    onChange={(e) => this.updateOrganisationState('billingCity', e)} 
                    value={this.state.organisation.billingCity} />
                ) : (
                    <span>{this.state.organisation.billingCity}</span> 
                )}
            </td>
            <td>
                {this.state.editing ? (
                    <input name="city" type="text" 
                    onBlur={(e) => this.saveOrganisation('billingAddress', e.target.value)}
                    onChange={(e) => this.updateOrganisationState('billingAddress', e)} 
                    value={this.state.organisation.billingAddress} />
                ) : (
                    <span>{this.state.organisation.billingAddress}</span> 
                )}
            </td>
            <td>
                {this.state.editing ? (
                    <input name="city" type="text" 
                    onBlur={(e) => this.saveOrganisation('billingPostalCode', e.target.value)}
                    onChange={(e) => this.updateOrganisationState('billingPostalCode', e)} 
                    value={this.state.organisation.billingPostalCode} />
                ) : (
                    <span>{this.state.organisation.billingPostalCode}</span> 
                )}
            </td>
            <td>
                {this.state.editing ? (
                    <ButtonGroup>
                        <Button color="primary" 
                        onClick={() => this.switchOrganisationState(true)} 
                        active={this.state.orgActiveState === true}>Active</Button>
                        <Button color="primary" 
                        onClick={() => this.switchOrganisationState(false)} 
                        active={this.state.orgActiveState === false}>Pause</Button>
                    </ButtonGroup>
                ) : <span>
                        {this.state.organisation.active ? 'Active' : 'Paused'}
                    </span>
                }
            </td>
            <td>
                <DeleteOrganisationButton organisationId={this.props.organisation._id}/>
            </td>
            <td>
                <Button onClick={this.switchEditing}>{this.state.editing ? 'Done' : 'Edit'}</Button>
            </td>
        </tr>
      ) : <Loading/>
    );
  }
}

AdminOrganisationComponent.defaultProps = {
  organisation : {},
};

AdminOrganisationComponent.propTypes = {
  organisation: PropTypes.object,
  loading: PropTypes.bool,
};

export default AdminOrganisationComponent;