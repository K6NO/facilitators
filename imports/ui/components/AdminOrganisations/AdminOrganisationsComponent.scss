@import '../../stylesheets/colors';
.AdminOrganisationsComponent {

    background: #ddd;
}