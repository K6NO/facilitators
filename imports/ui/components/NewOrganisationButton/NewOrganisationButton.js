/* eslint-disable max-len, no-return-assign */
import React from 'react';
import { Table, Button, Row, Col } from 'reactstrap';
import NewOrganisationEditorComponent from '../AdminOrganisations/NewOrganisationEditorComponent';

// import './NewOrganisationButton.scss';

class NewOrganisationButton extends React.Component {
  constructor(props){
    super(props);
    this.state = {
        isOpen : false,
    }
  }

  openEditor = () => {
      if(!this.state.isOpen) {
          this.setState({
              isOpen: true,
          });
      }
  }

  closeEditor = () => {
      this.setState({
          isOpen: false,
      })
  }

  render() {
    return ( 
        <div className="NewOrganisationButton">
            <Row>
                <Col>
                    <Button 
                        color="primary"
                        onClick={this.openEditor}>Add Organisation
                    </Button>
                </Col>
            </Row>
            {this.state.isOpen ? (
                <Row>
                    <Col>
                        <NewOrganisationEditorComponent closeCallback={this.closeEditor} />
                    </Col>
                </Row>
            ) : ''}
        </div>
    );
  }
}


export default NewOrganisationButton;