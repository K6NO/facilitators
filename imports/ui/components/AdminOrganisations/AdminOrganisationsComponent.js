/* eslint-disable max-len, no-return-assign */

import React from 'react';
import PropTypes from 'prop-types';
import { Table, Row, Col } from 'reactstrap';
import AdminOrganisationComponent from './AdminOrganisationComponent';
import './AdminOrganisationsComponent.scss';

class AdminOrganisationsComponent extends React.Component {
  constructor(props){
    super(props);
  }

  render() {
    const { organisations } = this.props;
    return (
        <div className="border AdminOrganisationsComponent container">
       
        {organisations.length ? (
            <Row>
                <Col>
                    <Table responsive>
                        <thead>
                        <tr>
                            <th>Name</th>
                            <th>Country</th>
                            <th></th>
                            <th></th>
                            <th></th>
                        </tr>
                        </thead>
                        <tbody>
                        {organisations.map(org => (
                            <AdminOrganisationComponent key={org._id} organisation={org} />
                        ))}
                        </tbody>
                    </Table>
                </Col>
            </Row>
        ) : <div>There are no organisations</div>}
        </div>
    );
  }
}

AdminOrganisationsComponent.defaultProps = {
  organisations : [],
};

AdminOrganisationsComponent.propTypes = {
  organisations: PropTypes.arrayOf(PropTypes.object),
  loading: PropTypes.bool,
};

export default AdminOrganisationsComponent;