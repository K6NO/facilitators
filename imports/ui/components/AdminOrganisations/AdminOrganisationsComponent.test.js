import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Button } from 'reactstrap';
import AdminOrganisationsComponent from './AdminOrganisationsComponent';
import NewOrganisationButton from './NewOrganisationButton';
import AdminOrganisationComponent from './AdminOrganisationComponent';

const mockOrgs = [
    {_id: '123'},
    {_id: '456'},
    {_id: '789'}
];
test('AdminOrganisationsComponent renders with ', () => {
    const component = renderer.create(
        <AdminOrganisationsComponent 
        organisations={mockOrgs} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('AdminOrganisationsComponent has NewOrganisationButton and AdminOrganisationComponents', () => {
    const shallowComponent = shallow(
        <AdminOrganisationsComponent
            organisations={mockOrgs} 
        />
    );
    expect(shallowComponent.find(NewOrganisationButton).length).toBe(1);
    expect(shallowComponent.find(AdminOrganisationComponent).length).toBe(3);
});
