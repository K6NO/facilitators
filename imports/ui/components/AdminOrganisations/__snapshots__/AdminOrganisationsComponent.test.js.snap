// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminOrganisationsComponent renders with  1`] = `
<div
  className="border AdminOrganisationsComponent container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="NewOrganisationButton"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <button
              aria-label={null}
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              Add Organisation
            </button>
          </div>
        </div>
        
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="table-responsive"
      >
        <table
          className="table"
        >
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th>
                Country
              </th>
              <th>
                City
              </th>
              <th>
                Address
              </th>
              <th>
                Postal code
              </th>
              <th />
              <th />
              <th />
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  Paused
                </span>
              </td>
              <td>
                <button
                  aria-label={null}
                  className="btn btn-danger"
                  onClick={[Function]}
                  type="button"
                >
                  Delete
                </button>
              </td>
              <td>
                <button
                  aria-label={null}
                  className="btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Edit
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  Paused
                </span>
              </td>
              <td>
                <button
                  aria-label={null}
                  className="btn btn-danger"
                  onClick={[Function]}
                  type="button"
                >
                  Delete
                </button>
              </td>
              <td>
                <button
                  aria-label={null}
                  className="btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Edit
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  
                </span>
              </td>
              <td>
                <span>
                  Paused
                </span>
              </td>
              <td>
                <button
                  aria-label={null}
                  className="btn btn-danger"
                  onClick={[Function]}
                  type="button"
                >
                  Delete
                </button>
              </td>
              <td>
                <button
                  aria-label={null}
                  className="btn btn-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Edit
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
